src/helpers/apiScripts.js

```diff
@@ -0,0 +1,100 @@
+import axios from "axios";
+
+const apiLink = "https://novi-backend-api-wgsgz.ondigitalocean.app/api/blogposts";
+const apiHeaders = {"novi-education-project-id": "e7334bb5-32f1-462b-a9d0-349bb821c0ba"};
+
+/*---api pull 1.1 ------------------------------------------------------------------------------------------------------*/
+async function pullHelperApi(setError, setApiData, toggleLoading) {
```

Comment:
Je hebt super goed bezig geweest met het extraheren van deze requests om je code op te schonen. Echter, omdat je telkens state gebruikt in deze context, doe je dit normaliter door een **custom hook** te maken in React. Die hook beheert de state en geeft deze telkens terug op basis van de API call. Kijk hier eens: http://youtube.com/watch?v=I2Bgi0Qcdvc (video) of als je meer van lezen houdt: https://dev.to/shaedrizwan/building-custom-hooks-in-react-to-fetch-data-4ig6